src/fs_io.cc

```diff
@@ -138,30 +155,13 @@ diskCombineWrites(_fde_disk *fdd)
         for (dwrite_q *q = fdd->write_q; q != nullptr; q = q->next)
             len += q->len - q->buf_offset;
 
-        dwrite_q *wq = (dwrite_q *)memAllocate(MEM_DWRITE_Q);
-
-        wq->buf = (char *)xmalloc(len);
-
-        wq->len = 0;
-
-        wq->buf_offset = 0;
-
-        wq->next = nullptr;
-
-        wq->free_func = cxx_xfree;
-
-        while (fdd->write_q != nullptr) {
-            dwrite_q *q = fdd->write_q;
-
+        const auto wq = new dwrite_q(len);
+        while (const auto q = fdd->write_q) {
             len = q->len - q->buf_offset;
             memcpy(wq->buf + wq->len, q->buf + q->buf_offset, len);
             wq->len += len;
```

Comment:
This PR code looks very buggy to me: memcpy() is writing past the allocated memory buffer and then incrementing `wq-len` past the allocated buffer space. If you have not tested this code, please do!

N.B. Official code does not have this bug because it initializes `wq->len` to zero rather than `len`.

Reply:
Good catch. Thank you. Fixed in 9cb4fbbda2